src/lib/Libifl/int_manager.c

```diff
@@ -172,8 +172,20 @@ PBSD_manager(int c, int rq_type, int command, int objtype, char *objname, struct
 						objname,
 						aoplp,
 						extend);
-			if (rc && objtype == MGR_OBJ_JOB && pbs_errno != PBSE_UNKJOBID)
-				break;
+
+			if (objtype == MGR_OBJ_JOB) {
+				if (rc == 0)
+					break;
```

Comment:
please address this comment

Reply:
> this changes logic, we were not breaking before with rc == 0, are you sure this is what you intended to do?

This was actually a bug in the earlier code. In case of a job/resv object if we get rc == 0 we should quit from the function. This is because we found the right server which serves the request. We don't need to iterate over other servers. This is actually a performance bug and I have fixed it in my latest commit. Please take a look at my latest commit.